src/ex_color/lightning_callbacks.py

```diff
@@ -0,0 +1,110 @@
+import logging
+from typing import Any
+
+import lightning as L
+import torch
+from lightning.pytorch.callbacks import Callback
+
+from ex_color.events import StepMetricsEvent
+from ex_color.record import MetricsRecorder
+
+log = logging.getLogger(__name__)
+
+
+class MetricsCallback(Callback):
+    """Lightning callback to replace the metrics recording functionality."""
+
+    def __init__(self):
+        super().__init__()
+        self.metrics_recorder = MetricsRecorder()
+
+    def on_train_batch_end(
+        self,
+        trainer: L.Trainer,
+        pl_module: L.LightningModule,
+        outputs: Any,
```

Comment:
Is it possible to use a dataclass for the outputs? I see they're unpacked below using a series of conditionals. Is that usual for Lightning?